oak/common/oak_sign.cc

```diff
@@ -35,50 +34,48 @@ using ::google::crypto::tink::Ed25519PublicKey;
 namespace oak {
 const char kPrivateKeyPemTag[] = "PRIVATE KEY";
 
-std::string generate_ed25519_key_pair() {
-  StatusOr<Ed25519PrivateKey> key_or = Ed25519SignKeyManager().CreateKey(Ed25519KeyFormat());
+Ed25519PrivateKey parse_key(const std::string& private_key) {
+  // Retrieve private and public keys from the input private key string
```

Comment:
Strictly speaking this is just parsing the private key, right? The public key is then derived from that. I guess it's the same, but the comment suggests that the input string contains two keys (which is also confusing by the use of PEM), but I think there is actually just one key, right?

edit: I think I'm getting to the bottom of this: in EdDSA the public key is just the suffix of the private key: https://en.wikipedia.org/wiki/EdDSA#Summary

here I am decoding both test keys in our repo from https://github.com/project-oak/oak/tree/main/examples/keys/ed25519 , and if you compare them you can see that the public one is indeed a suffix of the private one:

https://gchq.github.io/CyberChef/#recipe=PEM_to_Hex()&input=LS0tLS1CRUdJTiBQUklWQVRFIEtFWS0tLS0tCk1GTUNBUUV3QlFZREsyVndCQ0lFSU9zUWdHSzhwOVZQTGVuMk8zREQvZlh2RjNMeXdieE0zMnI0Y2VQN1h6bEgKb1NNRElRQi9qVklLVTIxSGlManEvWk82SFY5QXR1MzltcEd2V1VRMXFNSmIzYVBJL2c9PQotLS0tLUVORCBQUklWQVRFIEtFWS0tLS0t

https://gchq.github.io/CyberChef/#recipe=PEM_to_Hex()&input=LS0tLS1CRUdJTiBQVUJMSUMgS0VZLS0tLS0KZjQxU0NsTnRSNGk0NnYyVHVoMWZRTGJ0L1pxUnIxbEVOYWpDVzkyanlQND0KLS0tLS1FTkQgUFVCTElDIEtFWS0tLS0t

cc @ipetr0v

Reply:
The confusion is because what `openssl` calls `ed25519` private key is different from that in RFC 8032. This is from openssl (https://github.com/google/boringssl/blob/master/include/openssl/curve25519.h):
```
// Note that, unlike RFC 8032's formulation, our private key representation
// includes a public key suffix to make multiple key signing operations with the
// same key more efficient. The RFC 8032 private key is referred to in this
// implementation as the "seed" and is the first 32 bytes of our private key.
```
What we use and store is the 64-byte `openssl` private key. `Tink` calls it `Ed25519PrivateKey`, but it is in fact a key pair. In the Rust code we call it `KeyPair`. So I named the CPP class `KeyPair` too. 

I've clarified this in the comments. But I am open to revising the naming and the comments further :)

Reply:
So many standards :) thanks for looking into it; I think it is important that we have tests to ensure that we have cross-compatibility between key format and signing schemes, across languages, since otherwise I fear things may get out of sync very quickly. WDYT?

Reply:
I agree. I am looking into this. I'll have a closer look at the Rust code. I assumed it was the same 64-byte openssl format, but the one you shared above is longer, although it still has the public key as its suffix. I'll add more tests too.

Reply:
Does `tink` have functions to convert keys to PKCS#8?

Reply:
Hmmm... I am sure I replied to this question, but does not seem like GitHub saved it! 
No, Tink does not have support for PKCS#8.